examples/deserialize-transaction/src/example.ts

```diff
@@ -0,0 +1,403 @@
+log.info(decodedTransactionMessage.lifetimeConstraint, '[step 3] The transaction lifetime constraint');
+
+/**
+ * Here we can see that the lifetime constraint is actually a blockhash
+ * The `decompileTransactionMessage` call inspects the transaction and can distinguish
+ * between blockhash and nonce for us
+ * We can also narrow this type with typescript
+ */
+if ('blockhash' in decodedTransactionMessage.lifetimeConstraint) {
+    log.info(`[step 3] The transaction blockhash is ${decodedTransactionMessage.lifetimeConstraint.blockhash}`);
+}
+
+/**
+ * Note that the `lastValidBlockHeight` won't necessarily match that used when the transaction
+ * was first created. This is not encoded into the transaction, so we can't decode it back out
+ * By default `decodeTransactionMessage` and `decompileTransactionMessage` will fetch the current
+ * `lastValidBlockHeight` and use that
+ * But if you know the correct value, you can pass it like so:
+ *
+ * ```ts
+ * const decodedTransactionMessage = await decodeTransactionMessage(compiledTransactionMessage, rpc, {
+ *   lastValidBlockHeight: latestBlockhash.lastValidBlockHeight,
+ * })
+ * ```
+ */
+
+// We can also view the decompiled instructions here
+log.info(
+    { ...decodedTransactionMessage.instructions[0], data: '(removed for brevity)' },
+    '[step 3] The first decoded instruction',
+);
+
+/**
+ * We can see the programAddress is `11111111111111111111111111111111`
+ * We no longer need to look this up in a list of accounts from an index
+ *
+ * We also see the address of each account
+ * For the second account, we have its address, as well as the lookup table it comes from
+ * Decompiling the transaction message unpacks all the lookup accounts for us
+ */
+
+// We removed the data array for brevity when we logged the instructions, but let's take a look at it now
+log.info(decodedTransactionMessage.instructions[0].data, '[step 3] The data bytes of the first instruction');
+
+// This is opaque, it's just a byte array, but it encodes exactly what the instruction is actually doing
+// Let's look next at how we can make sense of it
+
+/**
+ * STEP 4: PARSING THE INSTRUCTIONS
+ * To understand what is actually happening in each instruction, we need to decode the data field
+ * We will do this by using the generated `@solana-program/system` client, which can decode data
+ * from the @solana/web3.js instruction data structure for the System program
+ * We know from the program address (11111111111111111111111111111111) that the first instruction
+ * is to the system program
+ * You can generate such a client for any Solana program using Kinobi
+ * See https://github.com/kinobi-so/kinobi for more information on Kinobi
+ */
+
+const firstInstruction = decodedTransactionMessage.instructions[0];
+// Narrow the type, an instruction must have data to be identified
+assertIsInstructionWithData(firstInstruction);
+const identifiedInstruction = identifySystemInstruction(firstInstruction);
```

Comment:
Instructions can have no data, though. There are plenty of instructions with no data that only require a list of accounts.

Reply:
Are there programs that have multiple different instructions that are distinguished by the types of accounts passed in? Or would such a program (similarly to the memo program) just be treated as having one instruction type? 

My assumption was that a program that just takes accounts would (like the memo program) not have an `identify` function, and would just have one instruction that you can parse.

Reply:
That's a good point. I guess it's probably extremely rare, but you _could_ do different things based on the number of accounts provided. This would be an anti-pattern though.

Reply:
Re-reading this line in the example, I think I was just confused by what it was describing. You're saying an instruction must have data to be identified _by the assert function_, not in general. You can disregard my comment here.